### PR TITLE
Update AndroidX dependencies.

### PR DESCRIPTION
**Pull request by pocmo**
We haven't updated them in a while and there's a new version of almost anyone available. With this I can build sample browser locally. Let's see what taskcluster finds.

### CODE REVIEW

**Comment by pocmo** [comment]
Of course it is red. :)

**Comment by pocmo** [comment]
```
[task 2020-10-29T11:15:52.817Z] w: /builds/worker/checkouts/src/samples/glean/src/androidTest/java/org/mozilla/samples/glean/MainActivityTest.kt: (11, 27): 'ActivityTestRule<T : Activity!>' is deprecated. Deprecated in Java
[task 2020-10-29T11:15:52.817Z] w: /builds/worker/checkouts/src/samples/glean/src/androidTest/java/org/mozilla/samples/glean/MainActivityTest.kt: (23, 23): 'ActivityTestRule<T : Activity!>' is deprecated. Deprecated in Java
[task 2020-10-29T11:15:52.817Z] w: /builds/worker/checkouts/src/samples/glean/src/androidTest/java/org/mozilla/samples/glean/MainActivityTest.kt: (23, 56): 'ActivityTestRule<T : Activity!>' is deprecated. Deprecated in Java
[task 2020-10-29T11:15:52.817Z] w: /builds/worker/checkouts/src/samples/glean/src/androidTest/java/org/mozilla/samples/glean/pings/BaselinePingTest.kt: (11, 27): 'ActivityTestRule<T : Activity!>' is deprecated. Deprecated in Java
[task 2020-10-29T11:15:52.817Z] w: /builds/worker/checkouts/src/samples/glean/src/androidTest/java/org/mozilla/samples/glean/pings/BaselinePingTest.kt: (62, 23): 'ActivityTestRule<T : Activity!>' is deprecated. Deprecated in Java
[task 2020-10-29T11:15:52.817Z] w: /builds/worker/checkouts/src/samples/glean/src/androidTest/java/org/mozilla/samples/glean/pings/BaselinePingTest.kt: (62, 56): 'ActivityTestRule<T : Activity!>' is deprecated. Deprecated in Java
```

**Comment by pocmo** [comment]
> task 2020-11-02T11:14:47.398Z] Installing collected packages: zipp, importlib-metadata, pyrsistent, attrs, jsonschema, Click, PyYAML, appdirs, MarkupSafe, Jinja2, diskcache, pathspec, yamllint, glean-parser
[task 2020-11-02T11:14:47.398Z]   Running setup.py install for pyrsistent: started
[task 2020-11-02T11:14:48.298Z]     Running setup.py install for pyrsistent: finished with status 'done'
[task 2020-11-02T11:14:48.398Z]   Running setup.py install for PyYAML: started
[task 2020-11-02T11:14:48.598Z]     Running setup.py install for PyYAML: finished with status 'done'
[task 2020-11-02T11:14:48.898Z] Successfully installed Click-7.1.2 Jinja2-2.11.2 MarkupSafe-1.1.1 PyYAML-5.3.1 appdirs-1.4.4 attrs-20.2.0 diskcache-5.0.3 glean-parser-1.28.6 importlib-metadata-2.0.0 jsonschema-3.2.0 pathspec-0.8.0 pyrsistent-0.17.3 yamllint-1.25.0 zipp-3.4.0
[task 2020-11-02T11:14:48.998Z] You are using pip version 10.0.1, however version 20.3b1 is available.
[task 2020-11-02T11:14:48.998Z] You should consider upgrading via the 'pip install --upgrade pip' command.
[task 2020-11-02T11:14:49.098Z] 
[task 2020-11-02T11:14:49.098Z] > Task :browser-engine-gecko:build_condas
[task 2020-11-02T11:14:49.098Z] > Task :browser-engine-gecko:build_conda_envs SKIPPED
[task 2020-11-02T11:14:49.098Z] > Task :browser-engine-gecko:build_pythons SKIPPED
[task 2020-11-02T11:14:49.098Z] > Task :browser-engine-gecko:build_pythons_from_zip SKIPPED
[task 2020-11-02T11:14:49.098Z] > Task :browser-engine-gecko:build_virtual_envs SKIPPED
[task 2020-11-02T11:14:49.098Z] > Task :browser-engine-gecko:build_envs
[task 2020-11-02T11:14:49.198Z] 
[task 2020-11-02T11:14:49.198Z] FAILURE: Build failed with an exception.
[task 2020-11-02T11:14:49.198Z] 
[task 2020-11-02T11:14:49.198Z] * What went wrong:
[task 2020-11-02T11:14:49.198Z] Execution failed for task ':support-sync-telemetry:Bootstrap_CONDA_'Miniconda3''.
[task 2020-11-02T11:14:49.198Z] > Process 'command 'bash'' finished with non-zero exit value 1

**Review by badboy — APPROVED** [review]
Looks good for Glean. We should probably updated that on our side as well (https://bugzilla.mozilla.org/show_bug.cgi?id=1674734)

**Comment by badboy** [comment]
Oh, I didn't see the second message with the build failure because of Miniconda.
Looping in @mdboom here.

**Comment by pocmo** [comment]
The Minoconda failure has happened on other PRs that tried to bump versions too, e.g. here:
* Updating Android Gradle plugin to 4.1: https://github.com/mozilla-mobile/android-components/pull/8761
* Updating Mockito to 3.5.13: https://github.com/mozilla-mobile/android-components/pull/8687

I do not understand why this is happening, those changes seem unrelated. Some help from the Glean team would be appreciated :)

**Comment by pocmo** [comment]
One potentially difference is that the toolchain task uses `--parallel` to run all the tasks at once (in order to cache all dependencies). Maybe that could cause some kind of race? Anyhow, just a guess.

**Comment by mdboom** [comment]
Yeah, I agree, it doesn't seem directly related and the log doesn't really provide any clues.  Your guess that it's a race seems as good as any.  Is it possible to install miniconda early and not in parallel to confirm the theory?

**Comment by pocmo** [comment]
> Is it possible to install miniconda early and not in parallel to confirm the theory?

@mdboom What would be the right way to do this? Is there some task we can run? Do we need to run it for only one module that uses it?

**Comment by mdboom** [comment]
So, perhaps there's a clue here: the task that's failing is `:support-sync-telemetry:Bootstrap_CONDA_'Miniconda3'`.  This suggests there is probably also a similar task for other Glean-using libraries in tree (such as Glean itself).  Since they all install the same miniconda environment to the same place, there would definitely be a problem if they run at the same time (though I believe running them sequentially should be fine and not duplicate work).

I am not at all a Gradle expert at all -- I wonder if there's a way to declare that the tasks of this type shouldn't run in parallel, or somehow restrict only the first one to actually doing work.  This is compounded a bit by this being a task from a third-party plugin from JetBrains that has had a host of other bugs we've had to workaround.  In any case, I'll investigate what Gradle tools are at our disposal here...

File: buildSrc/src/main/java/Dependencies.kt [hunk 1]
```diff
@@ -34,31 +34,32 @@ object Versions {
...
         const val browser = "1.2.0"
         const val cardview = "1.0.0"
-        const val constraintlayout = "1.1.3"
-        const val core = "1.2.0"
-        const val fragment = "1.1.0"
-        const val recyclerview = "1.0.0"
-        const val test = "1.2.0"
-        const val test_ext = "1.1.1"
-        const val espresso = "3.2.0"
+        const val constraintlayout = "2.0.3"
```
Comment (ekager) [inline buildSrc/src/main/java/Dependencies.kt:45]: They did a quick follow with 2.0.4 to fix an NPE and recommend using this version: https://developer.android.com/jetpack/androidx/releases/constraintlayout#2.0.4

Comment (pocmo) [reply buildSrc/src/main/java/Dependencies.kt:45]: Thanks! Will update this.

**Comment by pocmo** [comment]
Looks like it still happened with Glean 33.1.2. :(

File: docs/changelog.md [hunk 2]
```diff
@@ -16,7 +16,10 @@ permalink: /changelog/
   * 🚒 Bug fixed [issue #8464](https://github.com/mozilla-mobile/android-components/issues/8464) - Crash when confirming a prompt that was already confirmed
 
 * **feature-app-links**
-    * Added handling of PackageItemInfo.packageName NullPointerException on some Xiaomi and TCL devices
+  * Added handling of PackageItemInfo.packageName NullPointerException on some Xiaomi and TCL devices
+
+* **service-glean**
+  * 🆙  Updated Glean to version 33.1.2 ([changelog](https://github.com/mozilla/glean/releases/tag/v33.1.2))
```
Comment (jonalmeida) [inline docs/changelog.md:22]: I wonder if it's useful to add a general changelog entry saying something like, "Note that various AndroidX dependencies have been updated in this release."

**Review by jonalmeida — APPROVED** [review]
Thank you for doing this! 🙏 

I filed https://github.com/mozilla-mobile/android-components/issues/9088 as a follow-up to remove the deprecations from the `requireFragmentManager` usages.

**Pull request merged**